tests/core/test_device_collector.py

```diff
@@ -56,19 +61,69 @@ def test_device_connector_sync_run_engine_created_connects(RE):
     assert working_device.connected
 
 
+def test_device_connector_async_run_engine_same_event_loop():
```

Comment:
Not entirely sure from the name what this test is supposed to do, is it asserting that the device will use `loop` even if you call `checking_loop.run_until_complete`?

Reply:
I should change to `device_connector_async -> async_device_connector`... I'll clean it up.

It's checking that devices can be connected to in the `DeviceConnector` prior to any run-engine initialisation, and then can be put through the run-engine later so long as you explicitly pass in the same event-loop.

The test that's now skipping checks that the above fails if you don't pass in the same event-loop to the run-engine and let it create a new one. It doesn't pass currently, I think because `SimSignalBackends` will allow connection with one event-loop and `put` with another, which real devices will not.

Reply:
Why is the second checking loop required in that case?

Reply:
The `loop` can't be used because it's still active in the `RunEngine`. `An asyncio.run()` to get the motor setpoint works, but for some reason causes
`tests/core/test_device_save_loader.py::test_load_from_yaml` to fail. Happy to investigate why and address it in this PR.

I've noticed `tests/epics/test_signals.py::test_backend_get_put_monitor` is also timing out sometimes so maybe it would be best to make a new issue to fix these flakey tests.

Reply:
I'd just stick it in an issue for now